packages/oauth-providers/src/utils/getCodeChallenge.ts

```diff
@@ -0,0 +1,32 @@
+import CryptoJS from 'crypto-js'
+
+type Challenge = {
+  codeVerifier: string
+  codeChallenge: string
+}
+
+export function getCodeChallenge(): Challenge {
+  const codeVerifier = generateRandomString()
+  const codeChallenge = base64URLEncode(CryptoJS.SHA256(codeVerifier))
```

Comment:
Thanks for trying!

How about using the `Web Crypto` API directly? You can use `crypto.subtle.digest` function.

* https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto#Methods
* https://developers.cloudflare.com/workers/runtime-apis/web-crypto/#subtlecrypto-methods

It can depends external libraries since this is 3rd-party middleware, but if possible, we should not use it.

Reply:
Right! didn't thought about it